rust/main/agents/scraper/src/chain_scraper/mod.rs

```diff
@@ -153,43 +149,55 @@ impl HyperlaneSqlDb {
 
         for mut chunk in as_chunks::<(&H256, &mut (Option<i64>, i64))>(txns_to_fetch, CHUNK_SIZE) {
             for (hash, (_, block_id)) in chunk.iter() {
-                let info = self.provider.get_txn_by_hash(hash).await?;
+                let info = match self.provider.get_txn_by_hash(hash).await {
+                    Ok(info) => info,
+                    Err(e) => {
+                        warn!(?hash, ?e, "error fetching and parsing transaction");
+                        continue;
+                    }
+                };
                 hashes_to_insert.push(*hash);
                 txns_to_insert.push(StorableTxn {
                     info,
                     block_id: *block_id,
                 });
             }
 
+            // If we have no transactions to insert, we don't need to store them and update
+            // database transaction ids.
+            if txns_to_insert.is_empty() {
+                continue;
+            }
+
             self.db.store_txns(txns_to_insert.drain(..)).await?;
             let ids = self.db.get_txn_ids(hashes_to_insert.drain(..)).await?;
 
             for (hash, (txn_id, _block_id)) in chunk.iter_mut() {
-                let _ = txn_id.insert(ids[hash]);
+                ids.get(hash).map(|id| *txn_id = Some(*id));
```

Comment:
It will work in this case (not the other ones).

Below version leaves `txn_id` untouched if `ids` does not contain `hash`
```
                ids.get(hash).map(|id| *txn_id = Some(*id));
```

This version updates all `txn_id`, sometime with the same value of `None'.
```
                *txn_id = ids.get(hash).copied();
```

Reply:
Fixed.